src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -711,9 +711,35 @@ private void closeTempFile(String encodedPartition) {
   }
 
   private void closeTempFile() {
+    ConnectException connectException = null;
     for (String encodedPartition : tempFiles.keySet()) {
       // Close the file and propagate any errors
-      closeTempFile(encodedPartition);
+      try {
+        closeTempFile(encodedPartition);
+      } catch (ConnectException e) {
+        // still want to close all of the other data writers
+        connectException = e;
+        log.error(
+            "Failed to close temporary file for partition " + encodedPartition
+                + ". The connector will attempt to rewrite the temporary file."
+        );
+      }
+    }
+
+    if (connectException != null) {
+      // at least one tmp file did not close properly
```

Comment:
someone new to the code might wonder here, why don't we delete only the files that failed to close. 
IIUC that's because we won't be able to cherry-pick and skip the successful records on a retry (after we empty the buffer and reset the offsets). That would lead to duplicates. But should we add some info to this comment, so that we don't leave this completely to the reader's imagination?